src/Microsoft.Extensions.Configuration.AzureAppConfiguration/JsonConfiguration/JsonKeyValueAdapter.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure.Data.AppConfiguration;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.JsonConfiguration
+{
+    internal class JsonKeyValueAdapter : IKeyValueAdapter
+    {
+        public Task<IEnumerable<KeyValuePair<string, string>>> ProcessKeyValue(ConfigurationSetting setting, CancellationToken cancellationToken)
+        {
+            var keyValues = JsonConfigurationParser.ParseJsonSetting(setting);
+            return Task.FromResult<IEnumerable<KeyValuePair<string, string>>>(keyValues);
+        }
+
+        public bool CanProcess(ConfigurationSetting setting)
+        {
+            return JsonConfigurationParser.IsJsonContentType(setting?.ContentType);
```

Comment:
>IsJsonContentType [](start = 43, length = 17)

We can skip calling this method if `setting` is `null`.

Reply:
Done

---
In reply to: [454035078](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r454035078) [](ancestors = 454035078)